src/slc/java/com/lambdaherding/edi/slc/ch03/AdvancedEx1.java

```diff
@@ -0,0 +1,26 @@
+package com.lambdaherding.edi.slc.ch03;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class AdvancedEx1 {
+
+	public static List<Integer> mapStringToInteger(Stream<String> strings) {
+		return strings.reduce(new ArrayList<Integer>(), (ArrayList<Integer> a, String b) -> {
+			a.add(Integer.parseInt(b));
+			return a;
+		}, (ArrayList<Integer> a, ArrayList<Integer> b) -> {
+			return a;
+		});

```

Comment:
I think the exercise wanted this to implement map itself, e.g. accepting a mapper function and applying it to each element of the stream. This does still show the principle well enough though.